docs/release-process.md

```diff
@@ -0,0 +1,184 @@
+
+IMPORTANT: It _is_ permissible to manually edit the proposed changelog in the PR, but be aware those changes will be overwritten if another update is triggered. release-plz uses force-push to update any existing release PR.
+
+IMPORTANT: If the version number of the crate in `main` is _different_ from the version number on [crates.io](https://crates.io), this job will NOT run. (See the next task.)
+
+### Publish new crates
+
+For each crate in the repo, if the version number of the crate in `main` is _different_ from the version number on [crates.io](https://crates.io), release-plz will attempt to publish the crate to [crates.io](https://crates.io).
+
+Typically, this will happen because a maintainer merged a release PR created from the previous step. However, that is not absolutely required. ~~(Though **strongly discouraged,** you technically _could_ manually edit the version number in a `Cargo.toml` file and submit that directly to `main` yourself.)~~ (Turns out I'm wrong about this: release-plz has a specific safeguard that prevents the release process from running unless the last commit is from a release PR.)
+
+Specifically, it performs the following steps when the version number doesn't match what's on [crates.io](https://crates.io):
+
+* **Runs [`cargo publish`](https://doc.rust-lang.org/cargo/commands/cargo-publish.html)** for the crate and waits for notification that the crate has been successfully uploaded.
+* **Creates a [GitHub release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)** for the uploaded crate. This creates an annotated git tag of the form `(crate-name)-v(version-number)`, which – among other things – is used to drive the previous-version comparison for future release PRs. This also creates a ZIP archive of the source at this crate version which is stored on GitHub with the release.
+
+### Publish `c2patool` binary builds
+
+If a new version of `c2patool` is generated by the previous step, an additional job runs after the publish job to build and publish binary versions of `c2patool`.
+
+This is specified by the `publish-c2patool-binaries` job in our [`release.yml`](../.github/workflows/release.yml) task.
+
+For each supported command-line platform (MacOS, Windows, and Ubuntu), it performs the following steps:
+
+* **Runs the `release` task in [`cli/Makefile`](../cli/Makefile)**. This builds the platform-appropriate binary image.
+
+* **Runs [`cargo sbom`](https://crates.io/crates/cargo-sbom)** to generate an [SPDX-formatted](https://spdx.dev) software bill of materials (SBOM).
+
+* **Uploads the binary and SBOM files to GitHub.** These are attached as additional outputs to the GitHub release generated in the publish step above.
+
+## Related: Commit lint used for PR title enforcement
```

Comment:
I wonder if this shouldn't go on top, as introduction, since it's quite important also.

Reply:
Agreed and moved.